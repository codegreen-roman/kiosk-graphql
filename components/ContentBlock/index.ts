import KihnuKiosk from './themes/kihnuKiosk/ContentBlock'
import themeNames from '@const/themeNames'
import { chooseThemeComponent } from '@utils/theming'
import { FC, ReactNode } from 'react'

export interface ContentBlockProps {
  className?: string
  children: ReactNode
  selected?: boolean
  disabled?: boolean
}

export const ContentBlock: FC<ContentBlockProps> = chooseThemeComponent({
  [themeNames.kihnuKiosk]: KihnuKiosk,
})
