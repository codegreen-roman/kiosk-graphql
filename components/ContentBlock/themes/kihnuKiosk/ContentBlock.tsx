import React, { FC } from 'react'
import { Root } from './ContentBlockStyles'
import { ContentBlockProps } from '@components/ContentBlock'

const ContentBlock: FC<ContentBlockProps> = ({ className, children, selected, disabled }) => {
  return (
    <Root className={className} selected={selected} disabled={disabled}>
      {children}
    </Root>
  )
}

export default ContentBlock
