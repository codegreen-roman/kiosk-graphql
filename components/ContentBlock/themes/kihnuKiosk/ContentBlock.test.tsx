import ContentBlock from './ContentBlock'
import React from 'react'
import { render } from '@test/testUtils'
import 'jest-styled-components'
import { RenderResult } from '@testing-library/react'

describe('ContentBlock', () => {
  function renderComponent(): RenderResult {
    return render(
      <ContentBlock className="test" selected disabled>
        123
      </ContentBlock>
    )
  }

  it('renders correctly', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot()
  })
})
