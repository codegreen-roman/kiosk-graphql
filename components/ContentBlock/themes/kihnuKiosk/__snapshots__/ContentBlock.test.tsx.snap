// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBlock renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  background: #EDF2FF;
  border-color: #4266BD;
  border-width: 1px;
  border-style: solid;
  border-radius: 2px;
  padding: 11px 17px;
}

<div
    class="c0 test"
    disabled=""
  >
    123
  </div>
</DocumentFragment>
`;
